src/coord/src/coord.rs

```diff
@@ -2340,15 +2419,16 @@ impl Coordinator {
 
     /// A policy for determining the timestamp for a peek.
     ///
-    /// The result may be `None` in the case that the `when` policy cannot be satisfied,
-    /// which is possible due to the restricted validity of traces (each has a `since`
-    /// and `upper` frontier, and are only valid after `since` and sure to be available
-    /// not after `upper`).
+    /// The Timestamp result may be `None` in the case that the `when` policy
+    /// cannot be satisfied, which is possible due to the restricted validity of
+    /// traces (each has a `since` and `upper` frontier, and are only valid after
+    /// `since` and sure to be available not after `upper`). The set of indexes
+    /// used is also returned.
     fn determine_timestamp(
         &mut self,
         uses_ids: &[GlobalId],
         when: PeekWhen,
-    ) -> Result<Timestamp, CoordError> {
+    ) -> Result<(Timestamp, Vec<GlobalId>), CoordError> {
```

Comment:
commenting here for lack of a better space -- the way this all shakes out it seems like as long as the schema has at least one unmaterialized source trying to peek in a transaction will not work. Just want to confirm -- that is expected behavior right?

Reply:
This was a really good question and you are correct. I've opened #6651 to track this because I think it's bad behavior, but can be improved later.